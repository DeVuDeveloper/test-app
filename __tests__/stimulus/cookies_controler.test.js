import { screen } from '@testing-library/dom';
import { getHTML, setHTML, startStimulus } from './_stimulus_helper';
import CookiesController from '../../app/javascript/controllers/cookies_controller';

beforeEach(() => startStimulus('cookies', CookiesController));

test('fetches oven status and updates UI', async () => {
  await setHTML(`
    <div data-controller="oven-status" data-oven-id="123">
      <div data-oven-status-target="timer"></div>
      <button data-oven-status-target="button">
        Remaining time: 0 seconds
      </button>
      <div data-oven-status-target="ready"></div>
      <div data-oven-status-target="loader"></div>
    </div>
  `);

  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve({ ready: true, time_left: 0 }),
    })
  );

  await screen.findByRole('button', { name: 'Remaining time: 0 seconds' });

  expect(getHTML()).toMatchSnapshot();
});



