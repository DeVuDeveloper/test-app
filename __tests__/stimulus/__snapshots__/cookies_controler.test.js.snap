// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches oven status and updates UI 1`] = `
"<div data-controller="oven-status" data-oven-id="123">
      <div data-oven-status-target="timer"></div>
      <button data-oven-status-target="button">
        Remaining time: 0 seconds
      </button>
      <div data-oven-status-target="ready"></div>
      <div data-oven-status-target="loader"></div>
    </div>"
`;
